### PR TITLE
Implement negative scenarios for location entity in API

### LINKED ISSUE #2488: Add negative scenarios for Location create functionality in API
It is necessary to improve coverage and add few, most necessary negative scenarios to the code

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #2488 

```
nosetests tests/foreman/api/test_location.py -m test_negative_create
...
----------------------------------------------------------------------
Ran 3 tests in 22.866s

OK
```

### CODE REVIEW

File: tests/foreman/api/test_location.py [hunk 1]
```diff
@@ -227,6 +227,44 @@ def test_positive_delete(self):
             location.read()
 
     @tier1
+    def test_negative_create_with_name(self):
+        """Create new locations using invalid names only
```
Comment (abalakh) [inline tests/foreman/api/test_location.py]: Nitpick: ``Try to create new locations...```

File: tests/foreman/api/test_location.py [hunk 2]
```diff
@@ -227,6 +227,44 @@ def test_positive_delete(self):
...
+        """
+        name = gen_string('alphanumeric')
+        location = entities.Location(name=name).create()
+        self.assertEqual(location.name, name)
+        with self.assertRaises(HTTPError):
+            entities.Location(name=name).create()
+
+    @tier1
+    def test_negative_create_with_domain(self):
+        """Create new location using non-existent domain identifier
```
Comment (abalakh) [inline tests/foreman/api/test_location.py]: Same here

**Comment by abalakh** [comment]
ACK

File: tests/foreman/api/test_location.py [hunk 3]
```diff
@@ -227,6 +227,44 @@ def test_positive_delete(self):
             location.read()
 
     @tier1
+    def test_negative_create_with_name(self):
+        """Try to create new location using invalid names only
```
Comment (sthirugn) [inline tests/foreman/api/test_location.py]: For future PRs, please s/Try/Attempt - not a blocker for this PR.

**Comment by sthirugn** [comment]
ACK

**Pull request merged**